### PR TITLE
WL-4020 created Lessons component to show latest announcements in a div.

### PR DESCRIPTION
**Pull request by ouit0408**
Added new dialog for Announcements in ShowPage.html ,related variables
and function are added in 'SimplePageBean.java' to create/edit
SimplePageItem of type ANNOUNCEMENTS for Announcements summary widget.
'messages.properties' is edited to add new texts.

### CODE REVIEW

File: tool/src/webapp/js/announcements.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	showAnnouncements();
+});
+
+function showAnnouncements(){
+	var url = $('.announcements-site-url').text().replace(/'/g,"");
+	var announcementsUrl = url + ".json?n=" + number;
+
+	//Get announcements
+	$.ajaxSetup({ cache: false,async: true });	//Disable caching for ajax requests but make asynchronous
+	$.getJSON(announcementsUrl, function(data, status, xhr) {
```
Comment (buckett) [inline tool/src/webapp/js/announcements.js:21]: What about displaying something when the `.getJSON` function fails for some reason?

File: tool/src/webapp/js/announcements.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		var text_for_announcements = '<div class="announcementsHeaderDiv"><h3 class="announcementSummaryHeader" style="border-bottom:1px solid #ccc;padding-bottom: 5px;"><a href="'+link_to_tool+'" target="_top" title ="Announcements">Announcements</a></h3></div>';
+		if($(data["announcement_collection"]).size() === 0) {
+			//ie no announcements
+			text_for_announcements = '<p>'+msg("simplepage.announcements-no-message")+'</p>';
+		}
+		else {
+			$(data["announcement_collection"]).each(function(){
+				date = new Date(this["createdOn"]);//*1000);
+				var hour = date.getHours() < 10 ? '0' + date.getHours() : date.getHours();
+				date_time = hour +":"+date.getMinutes()+ " " + date.getDate() + "/"+ (date.getMonth()+1) + "/"+ date.getFullYear();// create a new javascript Date object based on the timestamp
```
Comment (buckett) [inline tool/src/webapp/js/announcements.js]: Is this all ok with relation to timezones? Sakai has the concept of a timezone, does the date returned by entitybroker include the user's timezone?

File: tool/src/webapp/js/announcements.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+				text_for_announcements += '<div class="itemDiv">';
+				var href = tool_href + this["announcementId"]+"&sakai_action=doShowmetadata";
+				text_for_announcements += '<div class="itemTitle"><a href="'+href+'" target="_top">'+this["entityTitle"]+'</a> by '+this["createdByDisplayName"] +'</div>';
```
Comment (buckett) [inline tool/src/webapp/js/announcements.js]: I think you need to escape HTML entities as the JSON returned will just be the raw title.

File: tool/src/webapp/js/announcements.js (hunk 2 above)
Comment (ouit0408) [inline tool/src/webapp/js/announcements.js]: Yes the date returned by entitybroker includes user's timezone.

**Comment by ouit0408** [comment]
Have implemented review comments and have updated PR.

**Comment by buckett** [comment]
Doesn't merge

File: tool/src/webapp/js/announcements.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		}
+		else {
+			$(data["announcement_collection"]).each(function(){
+				date = new Date(this["createdOn"]);//*1000);
+				var hour = date.getHours() < 10 ? '0' + date.getHours() : date.getHours();
+				var min = date.getMinutes() < 10 ? '0' + date.getMinutes() : date.getMinutes();
+				date_time = hour +":"+min+ " " + date.getDate() + "/"+ (date.getMonth()+1) + "/"+ date.getFullYear();// create a new javascript Date object based on the timestamp
+				text_for_announcements += '<div class="itemDiv">';
+				var href = tool_href + this["announcementId"]+"&sakai_action=doShowmetadata";
+				text_for_announcements += '<div class="itemTitle"><a href="'+href+'" target="_top">'+this["entityTitle"].escapeHTML()+'</a> by '+this["createdByDisplayName"].escapeHTML() +'</div>';
```
Comment (buckett) [inline tool/src/webapp/js/announcements.js:36]: This needs updating for i18n date formatting.

File: tool/src/webapp/js/announcements.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+				text_for_announcements += '<div class="itemDate">'+date_time+'</div>';
+				text_for_announcements += '</div>';
+			});
+		}
+		$('.announcements-div').html(text_for_announcements);
+	})
+	.fail(function( jqxhr, textStatus, error ) {	//if ajax request fails display error message
+		var err = textStatus + ", " + error;
+		text_for_announcements += '<p>Error in adding announcements : '+ err + '</p>';
```
Comment (buckett) [inline tool/src/webapp/js/announcements.js]: Need to load this from a resource bundle.

File: tool/src/webapp/js/announcements.js (hunk 4 above)
Comment (ouit0408) [inline tool/src/webapp/js/announcements.js:36]: As of now using user's locale and time zone setting for date formatting.

**Comment by ouit0408** [comment]
Made changes and have updated PR.

**Pull request merged**